apps/views.py

```diff
@@ -248,21 +248,21 @@ def post(self, request, user, project, ai_id):
 
         appinstance.name = request.POST.get("app_name")
         appinstance.description = request.POST.get("app_description")
+        appinstance.parameters.update(parameters)
+        appinstance.access = access
+        appinstance.save()
+        appinstance.app_dependencies.set(app_deps)
+        appinstance.model_dependencies.set(model_deps)
+        
         current_release_name = appinstance.parameters["release"]
         # if subdomain is set as --generated--, then use appname
         if request.POST.get("app_release_name") == "":
             new_release_name = appinstance.parameters["appname"]
         else:
             new_release_name = request.POST.get("app_release_name")
-        appinstance.parameters.update(parameters)
-        appinstance.access = access
-        new_url = appinstance.table_field["url"].replace(current_release_name, new_release_name)
-        appinstance.table_field.update({"url": new_url})
-        appinstance.save()
-        appinstance.app_dependencies.set(app_deps)
-        appinstance.model_dependencies.set(model_deps)
-        # check if new subdomain has been created
-        if current_release_name != new_release_name:
+        if new_release_name and current_release_name != new_release_name:
+            new_url = appinstance.table_field["url"].replace(current_release_name, new_release_name)
+            appinstance.table_field.update({"url": new_url})
```

Comment:
Honestly, this function is becoming a bit hard to read. It would be nice to make `update_appinstance` or something.

Just saying

Reply:
I agree! We should do clean up of many views files imo